### PR TITLE
Add support for custom grafana image for cephadm

### PR DESCRIPTION
**Pull request by psathyan**
# Description

The following changes are carried out in this PR

- Added support for custom grafana image test
- Workaround _Add host_ failure due to infra gaps with FQDN resolve error
- Fixed an issue with host add, the change is to check if the Object is none type before comparing the node list
- Modified the tier-0 test suite for deployment.

**Logs**
http://magna002.ceph.redhat.com/ceph-qe-logs/psathyan/fix_adm/result.html

### CODE REVIEW

File: ceph/ceph_admin/host.py [hunk 1]
```diff
@@ -63,16 +63,16 @@ def add(self, config):
         node = args.pop("node")
         ceph_node = get_node_by_id(self.cluster, node_name=node)
 
+        if not ceph_node:
```
Comment (sunilkumarn417) [inline ceph/ceph_admin/host.py:66]: Thanks for fixing this.

File: pipeline/5/Jenkinsfile-tier-0.groovy [hunk 2]
```diff
@@ -43,7 +43,7 @@ node(nodeName) {
         stage('Install prereq') {
             checkout([
                 $class: 'GitSCM',
-                branches: [[name: '*/master']],
+                branches: [[name: '*/fix_tier0']],
```
Comment (sunilkumarn417) [inline pipeline/5/Jenkinsfile-tier-0.groovy:46]: Are we shifting to tier-0 branch temporarily?

File: suites/pacific/cephadm/tier_0_cephadm.yaml [hunk 3]
```diff
@@ -497,6 +498,15 @@ tests:
               - node2
       destroy-cluster: false
       abort-on-fail: true
+  - test:
```
Comment (sunilkumarn417) [inline suites/pacific/cephadm/tier_0_cephadm.yaml:501]: As discussed, Can we take this in with `--grafana-image` cli option and update whenever needed for test-suites which have `-skip-monitoring-stack` options.

File: pipeline/5/Jenkinsfile-tier-0.groovy (hunk 2 above)
Comment (psathyan) [reply pipeline/5/Jenkinsfile-tier-0.groovy:46]: When we run the stages in parallel, I am observing the test execution time is at-least 10% more. Hence I am using this branch to experiment before I revert the code to `master`.

File: suites/pacific/cephadm/tier_0_cephadm.yaml [hunk 4]
```diff
@@ -498,6 +498,15 @@ tests:
...
+      name: configure custom grafana image
+      desc: Configure custom image for grafana
+      module: test_bootstrap.py
+      config:
+        command: shell
+        args:
+          - "ceph config set mgr mgr/cephadm/container_image_grafana"
+          - registry.redhat.io/rhceph-alpha/rhceph-5-dashboard-rhel8:5-15
```
[the remark this answers is not in the document]
Comment (psathyan) [reply suites/pacific/cephadm/tier_0_cephadm.yaml:509]: Thanks. I'll make the changes to use the `latest`

**Review by sunilkumarn417 — APPROVED** [review]
LGTM.

**Pull request merged**